arrows/core/derive_metadata.cxx

```diff
@@ -319,20 +376,26 @@ compute_derived_metadata( kwiver::vital::metadata_vector const& metadata_vec,
       updated_metadata->add(
         slant_range_trait.create_metadata_item( slant_range_value ) );
 
-      // Compute GSD
-      auto const gsd_value =
-        compute_gsd( updated_metadata, frame_width, frame_height );
-      auto const& gsd_trait = meta_traits.find( kv::VITAL_META_AVERAGE_GSD );
-      updated_metadata->add( gsd_trait.create_metadata_item( gsd_value ) );
-
-      // Compute VNIIRS
-      auto const rer_value = compute_rer();
-      auto const snr_value = compute_snr();
-      auto const vniirs_value = compute_vniirs( gsd_value, rer_value,
-                                                snr_value );
-      auto const& vniirs_trait = meta_traits.find( kv::VITAL_META_VNIIRS );
-      updated_metadata->add(
-        vniirs_trait.create_metadata_item( vniirs_value ) );
+      if( input_image )
+      {
+        auto const frame_width = input_image->width();
```

Comment:
okay for now but, as discussed, it would be much better in the future to add image width and height to the metadata and update video/image readers to provide that metadata so we don't have to read and pass an image around just for it's size.